### PR TITLE
Feature/o21 kb 14

### PR DESCRIPTION
**Pull request by manecipoglu**
This PR is only supposed to be for the back-end but after making it, I tested with the front-end and realized a lot of bugs and mistakes on the front-end. Therefore I included the fixes inside this PR too, very well knowingly that this is not the right way to go about this. 

One interesting thing to point out is that I realized history.push method coming from react-router-dom also takes optional data when you invoke the it. This is huge because it allows us to pass data between different pages even when they don't have parent-child relationship and we cannot pass props between them. I used this to update the Warehouses page so that it reflects the changes after we edit a warehouse. 

I also made a helper function to fetch data from an API on the front-end. I also changed patch to put after today's lecture.

Lastly, AddWarehouse and EditWarehouse components are quite similar to each other. If we have time, we can explore some options to merge it into a single component.

### CODE REVIEW

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 1]
```diff
@@ -18,29 +19,6 @@ class AddWarehouse extends Component {
     error: "",
   };
 
-  async componentDidMount() {
```
Comment (nilouhou) [inline client/src/pages/AddWarehouse/AddWarehouse.jsx:21]: so we don't need getData here?

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 2]
```diff
@@ -77,7 +55,18 @@ class AddWarehouse extends Component {
...
+        contact: {
+          name: this.state.contactName,
+          position: this.state.position,
+          phone: this.state.phoneNumber,
+          email: this.state.email,
+        },
+      };
       axios
         .post(
           `${API_URL}/warehouses/${this.props.match.params.id}`,
```
Comment (nilouhou) [inline client/src/pages/AddWarehouse/AddWarehouse.jsx:72]: oops, I just noticed the URL I put is incorrect! it should be ${API_URL}/warehouses/add

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 3]
```diff
@@ -97,7 +86,9 @@ class AddWarehouse extends Component {
     return (
       <main className="edit">
         <div className="edit__heading">
-          <i className="icon icon--left-arrow"></i>
+          <Link to="/">
```
Comment (nilouhou) [inline client/src/pages/AddWarehouse/AddWarehouse.jsx:89]: nice!

File: client/src/pages/AddWarehouse/AddWarehouse.jsx (hunk 1 above)
Comment (manecipoglu) [reply client/src/pages/AddWarehouse/AddWarehouse.jsx:21]: I thought AddWarehouse page did not need to make a get request to the API

File: client/src/pages/AddWarehouse/AddWarehouse.jsx (hunk 2 above)
Comment (manecipoglu) [reply client/src/pages/AddWarehouse/AddWarehouse.jsx:72]: Yes, just double checked! We are creating a new warehouse. We don't need the id. Good catch.

File: client/src/pages/AddWarehouse/AddWarehouse.jsx (hunk 1 above)
Comment (nilouhou) [reply client/src/pages/AddWarehouse/AddWarehouse.jsx:21]: you are, right. I could not comment on your last point, but agree with you, if we have time, I like the idea to make add/edit DRY

**Comment by nilouhou** [comment]
Seems good to me to merge @FadyaJoudah do I have a green light from you to merge?

**Comment by nilouhou** [comment]
I go ahead and merge this branch thank you @manecipoglu

**Pull request merged**